projects/driver_controls/src/analog_io.c

```diff
@@ -0,0 +1,78 @@
+#include <stddef.h>
+
+#include "analog_io.h"
+#include "adc.h"
+#include "gpio.h"
+#include "event_queue.h"
+#include "input_event.h"
+
+// Arbitrary thresholds for gas pedal
+#define COAST_THRESHOLD 1000
+#define DRIVE_THRESHOLD 3000
+
+// Analog device identifiers
+typedef enum {
+  ANALOG_GAS_PEDAL = 0,
+  NUM_ANALOG_IO_INPUTS
+} AnalogIODevices;
+
+// Lookup table for device ids based on ADC channel
+static AnalogIODevices s_analog_devices[] = {
+  [ADC_CHANNEL_11] = ANALOG_GAS_PEDAL
+};
+
+static void prv_input_callback(ADCChannel adc_channel, void *context) {
+  Event e;
+
+  adc_read_raw(adc_channel, &e.data);
+
+  switch (s_analog_devices[adc_channel]) {
+    case ANALOG_GAS_PEDAL:
+      if (e.data < COAST_THRESHOLD) {
+        e.id = INPUT_EVENT_GAS_BRAKE;
+      } else if (e.data > DRIVE_THRESHOLD) {
+        e.id = INPUT_EVENT_GAS_PRESSED;
+      } else {
+        e.id = INPUT_EVENT_GAS_COAST;
+      }
+      break;
+  }
+
+  event_raise(e.id, e.data);
+}
+
+// Obtain the ADC channel from the GPIO Address
+static ADCChannel prv_get_channel(GPIOAddress address) {
```

Comment:
This is something that should probably live in the ADC driver, because the calculations done here wouldn't necessarily be true on another platform.

(Also, I'd appreciate it if you added some documentation on where you found these offsets.)